server/ast/alter_role.go

```diff
@@ -15,17 +15,105 @@
 package ast
 
 import (
+	"errors"
 	"fmt"
+	"strings"
 
 	vitess "github.com/dolthub/vitess/go/vt/sqlparser"
 
 	"github.com/dolthub/doltgresql/postgres/parser/sem/tree"
+	pgnodes "github.com/dolthub/doltgresql/server/node"
 )
 
 // nodeAlterRole handles *tree.AlterRole nodes.
 func nodeAlterRole(node *tree.AlterRole) (vitess.Statement, error) {
 	if node == nil {
 		return nil, nil
 	}
-	return nil, fmt.Errorf("ALTER ROLE is not yet supported")
+	if len(node.Name) == 0 {
+		// The parser should make this impossible, but extra error checking is never bad
+		return nil, errors.New(`role name cannot be empty`)
+	}
+	options := make(map[string]any)
+	for _, kvOption := range node.KVOptions {
+		optionName := strings.ToUpper(string(kvOption.Key))
+		switch optionName {
+		case "BYPASSRLS":
+			options["BYPASSRLS"] = nil
+		case "CONNECTION_LIMIT":
+			switch value := kvOption.Value.(type) {
+			case *tree.DInt:
+				if value == nil {
+					options["CONNECTION_LIMIT"] = int32(-1)
+				} else {
+					// We enforce that only int32 values will fit here in the parser
+					options["CONNECTION_LIMIT"] = int32(*value)
+				}
+			case tree.NullLiteral:
+				options["CONNECTION_LIMIT"] = int32(-1)
+			default:
+				return nil, fmt.Errorf(`unknown role option value (%T) for option "%s"`, kvOption.Value, kvOption.Key)
+			}
+		case "CREATEDB":
```

Comment:
Might add a top-level comment that all these nils are markers and we only use the keys for them, no values